### ISSUE TITLE
Testing issue: hpctoolkit@2022.05.15%gcc@11.1.0

### ISSUE THREAD

**Issue by tldahlgren** [LLNL/spack#33121]
### Steps to reproduce the failure(s) or link(s) to test output(s)

@mwkrentel 

Stand-alone test failures were detected in the E4S PR pipeline stack for this spec.  More information is available at:

- Build:  https://cdash.spack.io/build/1816922
- Stand-alone tests: https://cdash.spack.io/viewTest.php?buildid=1816922

And

- Build: https://cdash.spack.io/build/1815883
- Stand-alone tests: https://cdash.spack.io/viewTest.php?buildid=1815883

### Error message

selection sort unit test:

```
Begin smoke test of selection sort, hpcrun, hpcstruct, hpcprof.

hpcrun -e REALTIME@5000 -t -o hpctoolkit-sort-measurements ./sort
FAIL: hpcrun failed
Segmentation fault (core dumped)
FAILED: Command exited with status 1:
'/tmp/root/spack-stage/spack-standalone-tests/fg5iuk7agdq27ld5c2cixurwrizykio5/hpctoolkit-2022.05.15-d2pqsyi/cache/hpctoolkit/tests/run-sort'
Begin smoke test of selection sort, hpcrun, hpcstruct, hpcprof.

hpcrun -e REALTIME@5000 -t -o hpctoolkit-sort-measurements ./sort
FAIL: hpcrun failed
Segmentation fault (core dumped)
...
```

### Information on your system or the test runner

Runner:  uo-pegasus
OS Release: 4.18.0-372.9.1.el8.x86_64 
Spack version: spack-0.19.0.dev0 (d9849fabcfe260e1817316921c0556d28fa9b52c) 
Repository revision: [d9849fabcfe260e1817316921c0556d28fa9b52c](https://github.com/spack/spack/commit/d9849fabcfe260e1817316921c0556d28fa9b52c)

And

Runner: uo-jupiter
OS Release: 5.4.48-xe-max 
Spack version: spack-0.19.0.dev0 (10720fbb4e277c97a262dbf8d51f6b93c27dab2a) 
Repository revision: [10720fbb4e277c97a262dbf8d51f6b93c27dab2a](https://github.com/spack/spack/commit/10720fbb4e277c97a262dbf8d51f6b93c27dab2a)

### Additional information

_Update_ (also in comment below): The E4S CI Stack (https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml) is the only one running the tests ATM  and it defines the image (ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01).

### General information

- [X] I have reported the version of Spack/Python/Platform/Runner
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded any available logs
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by mwkrentel** [comment]
@tldahlgren 
I tried this on an x86_64 RHEL 8.5 system with spack-built gcc 11.1.0.
Works for me.

I'm not familiar with the mechanics of the E4S builds.
I assume this is running in a docker container?

Can you point me to the container specs so I can run exactly the same
config that you're running.

**Comment by tldahlgren** [comment]
> I'm not familiar with the mechanics of the E4S builds. I assume this is running in a docker container?
> 
> Can you point me to the container specs so I can run exactly the same config that you're running.

That's correct.  The only stack that currently runs the tests is the E4S Stack (https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml) that defines the image (`ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01`).

**Comment by mwkrentel** [comment]
@tldahlgren 
I tried this in the ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01 container.
Again, works for me.

```
# spack test run hpctoolkit  
==> Spack test 4szw2w3vi3ud3vir3w6bt65rxbrs3kh7
==> Testing package hpctoolkit-2022.05.15-47bi2xf
============================= 1 passed of 1 specs ==============================
```

I didn't use the `spack.yaml` file with an environment, concretize,
etc.  I just built toolkit 2022.05.15 the usual way I build it.
But I looked at the specs in `spack.yaml` and there's nothing special.

Is there some way you could try this at another site?  Another machine?

Someway you could get the `sort` binary and a core file?
I don't have much to go on here.

**Comment by alalazo** [comment]
@tldahlgren I tried looking through the dashboards, but couldn't find any link back to the pipeline. Can you post one?

**Comment by tldahlgren** [comment]
> @tldahlgren I tried looking through the dashboards, but couldn't find any link back to the pipeline. Can you post one?

Unfortunately I couldn't find it either.  I'm working with Kitware to see if we can add the PR to the CDash output to make it easier to provide at least access to the artifacts.

**Comment by tldahlgren** [comment]
> @tldahlgren I tried this in the ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01 container. Again, works for me.
> 
> ```
> # spack test run hpctoolkit  
> ==> Spack test 4szw2w3vi3ud3vir3w6bt65rxbrs3kh7
> ==> Testing package hpctoolkit-2022.05.15-47bi2xf
> ============================= 1 passed of 1 specs ==============================
> ```
> 
> I didn't use the `spack.yaml` file with an environment, concretize, etc. I just built toolkit 2022.05.15 the usual way I build it. But I looked at the specs in `spack.yaml` and there's nothing special.
> 
> Is there some way you could try this at another site? Another machine?
> 
> Someway you could get the `sort` binary and a core file? I don't have much to go on here.

Apologies.  I tried to find the artifacts associated with the issues I reported that day.  We're working on a way to tie the CDash output to the PR so we can provide that information.

**Comment by scheibelp** [comment]
@scottwittenburg located some pipelines that built this same hash so could be useful for this:

* PR build: https://gitlab.spack.io/spack/spack/-/jobs/3574977
* develop build: https://gitlab.spack.io/spack/spack/-/jobs/3565886

From the first link, you can for example get the download url and do:

```
$ spack ci reproduce-build https://gitlab.spack.io/spack/spack/-/jobs/3574977/artifacts/download
==> Fetching artifacts from: https://gitlab.spack.io/spack/spack/-/jobs/3574977/artifacts/download

==> Job ran with the following image: ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01
checkout_commit: c959d6c905d26c2edcfbafa7311f09789b23b91a
merge_commit: 82ecb136a62e4b4e17a93d98f6d821ba9889e7ba
==> Error: Missing commit: 82ecb136a62e4b4e17a93d98f6d821ba9889e7ba
==> Error: Failed to automatically setup the tested version of spack in your local reproduction directory.

        This can happen if the spack you are using to run this command is not a git
        repo, or if it is a git repo, but it does not have the commits needed to
        recreate the tested merge commit.  If you are trying to reproduce a spack
        PR pipeline job failure, try fetching the latest develop commits from
        mainline spack and make sure you have the most recent commit of the PR
        branch in your local spack repo.  Then run this command again.
        Alternatively, you can also manually clone spack if you know the version
        you want to test.
            
==> Job ran with the following tags: ['spack', 'medium', 'x86_64', 'public']

Run the following command:

    $ docker run --rm -v /Users/scheibel1/Desktop/spack/spack/ci_reproduction:/builds/spack/spack -ti ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01

Once inside the container:

    - Clone spack and acquire tested commit

commit d9849fabcfe260e1817316921c0556d28fa9b52c
Merge: c959d6c9 82ecb136
Author: Sergey Kosukhin <sergey.kosukhin@mpimet.mpg.de>
Date:   Thu Oct 6 11:27:36 2022 +0000

    Merge 82ecb136a62e4b4e17a93d98f6d821ba9889e7ba into c959d6c905d26c2edcfbafa7311f09789b23b91a

    - Activate the environment

        $ source <spack-clone-path>/share/spack/setup-env.sh
        $ spack env activate --without-view /builds/spack/spack/jobs_scratch_dir/reproduction

    - Run the install script

        $ /builds/spack/spack/jobs_scratch_dir/reproduction/install.sh
```

I think the error can be ignored

(we agreed though it would be good for the test output to provide this link)

**Comment by scottwittenburg** [comment]
I don't think you should ignore the error.  As the explanation describes, you should try to fetch the latest commits from upstream (and if you're reproducing a job that ran on someone else PR, fetch the history from their fork too).  The `reproduce-build` command is trying to recreate the version of spack tested in the pipeline, and it does that by creating a new merge commit with the same parents as was tested in gitlab.

**Comment by mwkrentel** [comment]
@scheibelp I tried this, but there was no directory `/builds` in the container.

```
$ source <spack-clone-path>/share/spack/setup-env.sh
$ spack env activate --without-view /builds/spack/spack/jobs_scratch_dir/reproduction
```

**Comment by scheibelp** [comment]
I'll note that the download link from the second build works for me:

```
$ spack ci reproduce-build https://gitlab.spack.io/spack/spack/-/jobs/3565886/artifacts/download
```

so perhaps that will have a script in `/build/`.

@scottwittenburg when I pull the latest develop and retry the original link I posted, I get:

```
$ spack ci reproduce-build https://gitlab.spack.io/spack/spack/-/jobs/3574977/artifacts/download
==> Fetching artifacts from: https://gitlab.spack.io/spack/spack/-/jobs/3574977/artifacts/download

==> Error: while parsing a block mapping
  in "/Users/scheibel1/Desktop/spack/spack/ci_reproduction/spack/lib/spack/spack/test/data/config/compilers.yaml", line 3, column 5
expected <block end>, but found '{'
  in "/Users/scheibel1/Desktop/spack/spack/ci_reproduction/spack/lib/spack/spack/test/data/config/compilers.yaml", line 4, column 31
```

do you know why that would occur?
